controllers/sr_controller/sr_controller.py

```diff
@@ -176,7 +177,10 @@ def __exit__(
 
     def write(self, data):
         self.file.write(data)
-        self.stdout.write(data)
+        if data == '\n':
+            self.stdout.write(data)
+        else:
+            self.stdout.write(f'{self.zone}| {data}')
```

Comment:
This is a neat approach, however it won't necessarily work as you're hoping. Specifically it's possible that `write` will be called without a preceding newline having been emitted. This means that code like this:
``` python
print('foo', end='')
print('bar')
```
which we'd want to output "0 | foobar\n" would end up outputting "0| foo0| bar\n".